### PR TITLE
Extracted Payment to AppPayment package.

### PR DESCRIPTION
**Pull request by stereomon**
## PR Description

Initial commit

## Checklist
- [ ] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+/src/Orm/Zed/*/Persistence/Base
+/src/Orm/Zed/*/Persistence/Map
+/src/Orm/Propel/*
+!/src/Orm/Propel/.gitkeep
+
+/src/Generated
+
+/tests/*/*/*/_output/*
+/tests/*/*/*/_support/_generated/*
```
Comment (matweew) [inline .dockerignore:27]: remove file, not relevant for the package.

File: .git.docker [hunk 2]
```diff
@@ -0,0 +1 @@
+478b8cdea5e429097c28ed80658fa939fc83a1a6
```
Comment (matweew) [inline .git.docker:1]: remove file, not relevant for the package.

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[submodule "docker"]
+  path = docker
+  url = git@github.com:spryker/docker-sdk.git
```
Comment (matweew) [inline .gitmodules:3]: remove file, not relevant for the package.

File: .license [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/**
+ * This file is part of the Spryker Suite.
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
```
Comment (matweew) [inline .license:4]: not sure that is correct text, let's remove this file now and also this text from all php files.

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,40 @@
+SPRYKER SYSTEMS GMBH EVALUATION LICENSE AGREEMENT
```
Comment (matweew) [inline LICENSE:1]: has to be removed or clarified with legal.

File: README.md [hunk 6]
```diff
@@ -1 +1,71 @@
-# app-payment
+# AppPayment Module
```
Comment (matweew) [inline README.md:1]:
```suggestion
# AppPayment Package
```

because contains more than one module

File: config/Shared/config_default.php [hunk 7]
```diff
@@ -0,0 +1,173 @@
+<?php
+
+/**
+ * This configuration is used for TESTING only and will never be used in production!
+ */
```
Comment (matweew) [inline config/Shared/config_default.php:5]: can it be moved under tests/ folder?

File: config/Shared/config_default.php [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+    PaymentCancellationFailedTransfer::class => 'payment-events',
+    CancelPaymentTransfer::class => 'payment-commands',
+    CapturePaymentTransfer::class => 'payment-commands',
+    RefundPaymentTransfer::class => 'payment-commands',
+    AddPaymentMethodTransfer::class => 'payment-method-commands',
+    DeletePaymentMethodTransfer::class => 'payment-method-commands',
+    PaymentCreatedTransfer::class => 'payment-events',
+    // App event
+    AppConfigUpdatedTransfer::class => 'app-events',
+];
```
Comment (matweew) [inline config/Shared/config_default.php:156]: has to be added to Readme file so app developer knows what has to be added to config_default.php of the app.

File: src/Orm/Propel/Migration_SQLite/PropelMigration_1719313206.php [hunk 9]
```diff
@@ -0,0 +1,193 @@
+<?php
+use Propel\Generator\Manager\MigrationManager;
+
+/**
+ * Data object containing the SQL and PHP code to migrate the database
+ * up to version 1719313206.
+ * Generated on 2024-06-25 11:00:06 by reneklatt */
+class PropelMigration_1719313206{
```
Comment (matweew) [inline src/Orm/Propel/Migration_SQLite/PropelMigration_1719313206.php:8]: is it possible to delete migrations from the package?

File: src/Orm/Zed/AppKernel/Persistence/SpyAppConfig.php [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Skeleton subclass for representing a row from the 'spy_app_config' table.
+ *
+ *
+ *
+ * You should add additional methods to this class to meet the
+ * application requirements.  This class will only be generated as
+ * long as it does not already exist in the output directory.
+ */
+class SpyAppConfig extends BaseSpyAppConfig
```
Comment (matweew) [inline src/Orm/Zed/AppKernel/Persistence/SpyAppConfig.php:16]: if ORM classes (from other packages) are needed only for CI, let's generate them in CI

File: src/Spryker/Glue/AppPaymentBackendApi/AppPaymentBackendApiDependencyProvider.php [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+        return $container;
+    }
+
+    protected function addAppPaymentFacade(Container $container): Container
+    {
+        $container->set(static::FACADE_APP_PAYMENT, static function (Container $container): AppPaymentBackendApiToAppPaymentFacadeInterface {
+            // The PaymentFacade will always be mocked
+            // @codeCoverageIgnoreStart
+            return new AppPaymentBackendApiToAppPaymentFacadeBridge($container->getLocator()->appPayment()->facade());
```
Comment (matweew) [inline src/Spryker/Glue/AppPaymentBackendApi/AppPaymentBackendApiDependencyProvider.php:38]: this is one repository, so the bridge is overkill here, just use facade via locator.

File: src/Spryker/Zed/AppPayment/AppPaymentConfig.php [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+use Spryker\Shared\Application\ApplicationConstants;
+use Spryker\Shared\AppPayment\AppPaymentConstants;
+use Spryker\Shared\GlueJsonApiConvention\GlueJsonApiConventionConstants;
+use Spryker\Zed\Kernel\AbstractBundleConfig;
+
+class AppPaymentConfig extends AbstractBundleConfig
+{
+    /**
+     * @api
+     *
```
Comment (matweew) [inline src/Spryker/Zed/AppPayment/AppPaymentConfig.php:19]:
```suggestion
```

File: composer.json [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+    "require": {
+        "php": ">=8.1",
+        "spryker/app-webhook": "dev-main",
+        "spryker/glue-application-extension": "^1.0.0",
+        "spryker/kernel": "^3.30.0",
+        "spryker/log": "^3.0.0",
+        "spryker/message-broker": "^1.11.0",
+        "spryker/propel-encryption-behavior": "^0.1.1",
+        "spryker/secrets-manager": "^1.0.0",
+        "spryker/secrets-manager-extension": "^1.0.0",
```
Comment (matweew) [inline composer.json:15]: not used in the package, please validate other dependencies including req-dev

File: src/Spryker/Zed/AppPayment/Presentation/OnboardMerchant/onboarding.twig [hunk 14]
```diff
@@ -0,0 +1,7 @@
+{#This page needs to be implemented in the Stipe module same as we did with the payment page.#}
+
+<h1>Merchant Onboarding page</h1>
```
Comment (matweew) [inline src/Spryker/Zed/AppPayment/Presentation/OnboardMerchant/onboarding.twig:3]: to remove

File: src/Spryker/Zed/AppPayment/Presentation/OnboardMerchant/index.twig [hunk 15]
```diff
@@ -0,0 +1,17 @@
+{#This page needs to be implemented in the using module same as we did with the payment page.#}
+
+<h1>Merchant Onboarding page</h1>
```
Comment (matweew) [inline src/Spryker/Zed/AppPayment/Presentation/OnboardMerchant/index.twig:3]: to remove

**Review by matweew — CHANGES_REQUESTED** [review]

File: src/Spryker/Glue/AppPaymentBackendApi/AppPaymentBackendApiDependencyProvider.php (hunk 11 above)
Comment (stereomon) [reply src/Spryker/Glue/AppPaymentBackendApi/AppPaymentBackendApiDependencyProvider.php:38]: As for the others let's keep.

File: src/Orm/Zed/AppKernel/Persistence/SpyAppConfig.php (hunk 10 above)
Comment (stereomon) [reply src/Orm/Zed/AppKernel/Persistence/SpyAppConfig.php:16]: Committed in suite as well.

File: src/Spryker/Zed/AppPayment/AppPaymentConfig.php (hunk 12 above)
Comment (stereomon) [reply src/Spryker/Zed/AppPayment/AppPaymentConfig.php:19]: Auto added by CS

File: src/Orm/Zed/AppKernel/Persistence/SpyAppConfig.php (hunk 10 above)
Comment (matweew) [reply src/Orm/Zed/AppKernel/Persistence/SpyAppConfig.php:16]: it's not a suite (project code), it's a library package that cannot be used as a standalone project.

File: src/Spryker/Zed/AppPayment/AppPaymentConfig.php [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+     * @var string
+     */
+    public const HEADER_TENANT_IDENTIFIER = 'x-tenant-identifier';
```
Comment (matweew) [inline src/Spryker/Zed/AppPayment/AppPaymentConfig.php:22]: let's move to the business model where it's used.

**Comment by matweew** [comment]
if it's a library there must not be any `composer.lock`

File: LICENSE (hunk 5 above)
Comment (matweew) [reply LICENSE:1]: confirmed with Anneliese Hartlaub that we need to use this license.

**Pull request merged**